### DISCUSSION TITLE
how does FLAML work for forecasting?

### DISCUSSION THREAD

**Discussion by Yasslight90** [microsoft/FLAML#410] [General] [answered]
Hello,

I have been wondering how FLAML works for forecasting time series. I know it trains three models with different configurations: ARIMA, SARIMA and Prophet (GAM) but I don't know what happens next:

1- Does it select the most precise model on a validation set and use it for forecasting?
2- Does it combine the models to create an ensemble model and use it for forecasting?

Correct me please.

Have a nice day!

Yassine

**Comment by int-chaos** [comment, marked as answer]
Like you said FLAML will train the learner estimators with different configurations. Currently it supports ARIMA, SARIMA, Prophet and several sklearn regressors (LightGBM, xgboost, extra trees, and random forest). Given the inputted time budget, it will perform multiple iterations until the time runs out and optimize the hyperparameters accordingly. Each trained model is validated using a validation set that is the same length as the `period` parameter. The evaluation metric used is mape. When using FLAML to predict, it will use the model that performed the best on the validation set, which is the one with the lowest mape.

**Reply by torronen** [reply]
@int-chaos Do you have plans for adding in-built support for NeuralProphet?

**Reply by int-chaos** [reply]
Thank you for the suggestion. I haven't heard of it, but will certainly look into it and assess the feasibility of implementation. It sounds like a deep learning version of Prophet which sounds extremely interesting.

**Reply by torronen** [reply]
Yes, it is Prophet with added neural network. It seems fairly similar to Prophet API,  so I hope it should be only about adding parameter to enable AR-Net layer. https://github.com/ourownstory/neural_prophet

**Reply by shyam314159** [reply]
am a little bit unfulfilled with "period" parameter. can you please explain the significance of period parameter?
